documents/sd/chapter/Classes/BrowserExtension/Listeners/Listeners.tex

```diff
@@ -318,11 +354,11 @@ \subsection{Listeners}
 
 \clsdcl{class DownloadEventFactory}
 
-\clsdsp{The TabEventFactory is responsible for creating DownloadEvent objects from the information a DownloadListener gathers.}
+\clsdsp{The DownloadEventFactory is responsible for creating DownloadEvent objects from the information a DownloadListener gathers.}
 
 \begin{methods}
-\begin{method}{createEvent(downloadItem: downloads.DownloadItem) : DownloadEvent}{Create a new DownloadEvent from a DownLoadItem.}
-\return{BrowserEvent}{A BrowserEvent.}
+\begin{method}{createEvent(downloadItem: downloads.DownloadItem) : Promise<DownloadEvent>}{Asynchronously create a new DownloadEvent from a downlaods.DownLoadItem}
```

Comment:
Typo/This should probably be "from a download.DownloadItem" (unless the items are called "DownLoadItem", but I don't think that is the case, because you use "DownloadItem" in the signature)